### DISCUSSION TITLE
ScriptReference/Renderer-bounds

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#580] [Notes]
# ScriptReference/Renderer-bounds

https://docs.unity3d.com/ScriptReference/Renderer-bounds.html

**Comment by electronics45** [comment]
When using this method on a Tilemap Renderer be aware that the bounds of a tilemap grow automatically, but never shrink.  If you need accurate bounds for your project you need to manually call CompressBounds() on the tilemap, and this is an expensive operation.  This will work if you only have a handle to the Tilemap Renderer:
`renderer.GetComponentInParent <Tilemap> ()?.CompressBounds ();`

You could also manually select the "Compress Bounds" context menu action of the Tilemap component.

Further reading: https://forum.unity.com/threads/tilemap-bounds-bug.499738/
